test/jdk/javax/swing/ToolTipManager/bug5078214.java

```diff
@@ -0,0 +1,147 @@
+ * @run main bug5078214
+ */
+
+import java.awt.BorderLayout;
+import java.awt.event.MouseEvent;
+import java.awt.GraphicsConfiguration;
+import java.awt.GraphicsDevice;
+import java.awt.GraphicsEnvironment;
+import java.awt.Insets;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Toolkit;
+import java.awt.Window;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.ToolTipManager;
+
+public class bug5078214 {
+    static volatile boolean passed = false;
+
+    static volatile JFrame mainFrame;
+    static volatile Rectangle bounds;
+    static volatile Insets insets;
+    static volatile JRobot r;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                GraphicsEnvironment ge = GraphicsEnvironment.
+                        getLocalGraphicsEnvironment();
+                GraphicsDevice[] gs = ge.getScreenDevices();
+                GraphicsConfiguration testGraphics = null;
+                for (int j = 0; j < gs.length; j++) {
+                    GraphicsDevice gd = gs[j];
+                    GraphicsConfiguration[] gc =
+                            gd.getConfigurations();
+                    for (int i = 0; i < gc.length; i++) {
+                        insets = Toolkit.getDefaultToolkit().getScreenInsets(gc[i]);
+                        if (insets.bottom != 0) {
+                            testGraphics = gc[i];
+                            break;
+                        }
+                    }
+                }
+
+                if (testGraphics != null) {
+                    mainFrame = new JFrame("bug5078214");
+                    mainFrame.setLayout(new BorderLayout());
+                    mainFrame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+                    JButton button = new JButton("Button with tooltip") {
+                        public Point getToolTipLocation(MouseEvent ev) {
+                            return new Point(100, 100);
+                        }
+                    };
+                    mainFrame.add(button, BorderLayout.CENTER);
+                    button.setToolTipText("ToolTip for this button");
+
+                    // Position frame
+                    mainFrame.setSize(200, 200);
+                    bounds = testGraphics.getBounds();
+                    Insets insets = Toolkit.getDefaultToolkit()
+                            .getScreenInsets(testGraphics);
+                    int x = bounds.x + 200;
+                    int y = bounds.y + bounds.height - insets.bottom - 100;
+                    mainFrame.setLocation(x, y);
+                    mainFrame.setVisible(true);
+
+                } else {
+                    System.out.print("We need at least one screen with ");
+                    System.out.println("the taskbar at the bottom position.");
+                    System.out.println("Testing skipped.");
```

Comment:
As @azvegint mentioned you need either a return or jtreg Skipped Exception here.

Reply:
I would also recommend inverting the condition to avoid additional indentation.